source/precharge/main.c

```diff
@@ -125,24 +127,35 @@ int main (void)
 
     PHAL_writeGPIO(SPI_CS_ACEL_GPIO_Port, SPI_CS_ACEL_Pin, 0);
     PHAL_writeGPIO(SPI_CS_GYRO_GPIO_Port, SPI_CS_GYRO_Pin, 1);
-    
+
     NVIC_EnableIRQ(CAN1_RX0_IRQn);
     NVIC_EnableIRQ(CAN2_RX0_IRQn);
 
     /* Module init */
     schedInit(APB1ClockRateHz * 2); // See Datasheet DS11451 Figure. 4 for clock tree
     initCANParse(&q_rx_can);
-    BMS_init();
+    // Commented this line out because it is initializing old BMS, which is of no use to us
+    //BMS_init();
 
     /* Task Creation */
     schedInit(SystemCoreClock);
+    //Preflight animation + checks - not relavant to BMS
     configureAnim(preflightAnimation, preflightChecks, 75, 750);
+    //Just heartbeat
     taskCreate(heartbeatTask, 500);
+    //Monitors status of IMD, and then reflects this onto BMS. This is where I added Orion's error checks.
     taskCreate(monitorStatus, 50);
-    taskCreate(BMS_txBatteryStatus, 50);
-    taskCreate(BMS_chargePeriodic, 50);
+    /*
+        These are MiniBMS functions; Not sure whether to modify these or not, but I think they can just be completely removed.
+        However, I am not sure about other functions that rely on data from here.
+     */
+    // taskCreate(BMS_txBatteryStatus, 50);
+    // taskCreate(BMS_chargePeriodic, 50);
```

Comment:
We still need charging functionality. We will need to combine the Orion's charging protocol with that of the Elcon and make it controlled via DAQ. It will likely look similar to this function, ensure it takes into account temperature and put in plenty of error checking. Possibly consider putting the function you create in the Orion module.